### PR TITLE
Improve docs for FileSystemAccessControl impersonation rule 

### PR DESCRIPTION
**Pull request by sbernauer**
Extracted from https://github.com/trinodb/trino/pull/10500

### CODE REVIEW

File: docs/src/main/sphinx/security/file-system-access-control.rst [hunk 1]
```diff
@@ -364,29 +364,29 @@ use the following rules:
 Impersonation rules
```
Comment (kokosing) [inline docs/src/main/sphinx/security/file-system-access-control.rst:364]: Let's remove `doc: ` commit message prefix. We don't have such convention.

File: docs/src/main/sphinx/security/file-system-access-control.rst [hunk 2]
```diff
@@ -385,6 +385,8 @@ Each impersonation rule is composed of the following fields:
 * ``new_user`` (required): regex to match against the user that will be impersonated.
 * ``allow`` (optional): boolean indicating if the authentication should be allowed. Defaults to ``true``.
 
+The impersonation rules are a bit different than the other rules: The attribute new_user is required to not accidentally prevent more access than intended. Doing so it was possible to make the attribute allow optional.
```
Comment (kokosing) [inline docs/src/main/sphinx/security/file-system-access-control.rst:388]:
```
The attribute ``new_user``
```

Same for `allow`

**Review by kokosing — APPROVED** [review]
% comments

@mosabua Can you please take a look?

File: docs/src/main/sphinx/security/file-system-access-control.rst (hunk 1 above)
Comment (sbernauer) [reply docs/src/main/sphinx/security/file-system-access-control.rst:364]: Thanks, fixed

**Comment by kokosing** [comment]
Merged, thanks!

**Comment by sbernauer** [comment]
Thanks for merging!

**Pull request merged**